### PR TITLE
Update article comment form ids

### PR DESCRIPTION
**Pull request by cshold**
A pending PR in Shopify (https://github.com/Shopify/shopify/pull/57450) will reload the page after a comment form submission with either `#comments` or `#comment-form` as a hash in the URL.

- `#comments` already works as expected on a successful post
- `#comment-form` will be added when the form has errors, jumping the page down to the anchor on the form so the user can fix the mistakes

This PR removes the hacky JS that got around having accurate URL hashes and updates the IDs.

cc @Shopify/themes-team @mac-adam-chaieb 

FYI @Shopify/theme-support, this can be a reference for updating other themes. Ping me with any questions.

### CODE REVIEW

File: templates/article.liquid [hunk 1]
```diff
@@ -128,15 +128,12 @@
           {% endif %}
 
           {% comment %}
-            Comment submission form
+            #comment-form is required, it is used as an anchor link by Shopify
+            when an error occurs during submission.
           {% endcomment %}
-          <div class="form-vertical">
+          <div id="comment-form" class="form-vertical">
```
Comment (mac-adam-chaieb) [inline templates/article.liquid]: please remove this, as `{% form 'new_comment', article %} will dynamically generate a form with `id="comment_form"`

File: templates/article.liquid [hunk 2]
```diff
@@ -128,15 +128,12 @@
...
             {% form 'new_comment', article %}
```
Comment (mac-adam-chaieb) [inline templates/article.liquid]: this will take care of generating the `id="comment_form"`, as explained in the note above :)

Comment (suture) [inline templates/article.liquid]: The link to the PR is broken

Comment (cshold) [inline templates/article.liquid]: It's a private Shopify PR so you won't have access to it. Posted it so other Shopifolk would have a reference.

**Comment by suture** [comment]
Just a small detail :)

It seems the hash that gets added is #comment_form not #comment-form

**Comment by suture** [comment]
Also in blog.liquid you need to update the Comments link (in the meta data) which has the old hash #Comments

**Comment by carolineschnapp** [comment]
@suture 

>  Also in blog.liquid you need to update the Comments link (in the meta data) which has the old hash #Comments

Good one, thanks so much! We'll fix that.

We fixed `#comment-form` to `#comment_form` in the above comment.

**Comment by cshold** [comment]
Good catch, thanks @suture

**Comment by suture** [comment]
Maybe another problem. If you delete this, we no longer get a success message. 

```html
{% if form.posted_successfully? %}
      {% assign newHash = 'Comments' %}
 {% endif %}
```

**Comment by cshold** [comment]
Will be adding that in another PR shortly

**Comment by suture** [comment]
Many thanks for following that up.

**Pull request merged**